From afa9a32db7f9423b45e0cff898a868de05ad3dee Mon Sep 17 00:00:00 2001
From: Andrei Dinu <andrei.adrianx.dinu@intel.com>
Date: Tue, 30 Jul 2013 12:05:39 +0300
Subject: [PATCH] Curses and IOLoader changes

The linux distribution couldn't be identified when
running Bastille, and the question pruning method
couldn't get a match on the questions relevant to
the repo, so it eliminated all questions.

After answering the questions the checkAndSaveConfig routine
was called which was missing. Replaced it with outputConfig
which exists.

Signed-off-by: Andrei Dinu <andrei.adrianx.dinu@intel.com>
---
 Bastille/IOLoader.pm |    2 +-
 Bastille_Curses.pm   |    3 ++-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/Bastille/IOLoader.pm b/Bastille/IOLoader.pm
index abb94d7..995d2c2 100644
--- a/Bastille/IOLoader.pm
+++ b/Bastille/IOLoader.pm
@@ -68,7 +68,7 @@ sub Load_Questions($) {
   my $UseRequiresRules = $_[0];
 
   my ($current_module_number,$first_question) = &parse_questions();
-  $first_question = &prune_questions($UseRequiresRules,$first_question);
+  #$first_question = &prune_questions($UseRequiresRules,$first_question);
   $firstQuestion = $first_question;
   &B_log("DEBUG","Load Questions, first question: $first_question");
   &validate_questions();
diff --git a/Bastille_Curses.pm b/Bastille_Curses.pm
index 2e1eef4..a2dfb34 100644
--- a/Bastille_Curses.pm
+++ b/Bastille_Curses.pm
@@ -84,7 +84,8 @@ sub do_Bastille {
     }
 
     # Output answers to the script and display
-    &checkAndSaveConfig(&getGlobal('BFILE', "config"));
+
+    &outputConfig
 
     # Run Bastille
 
-- 
1.7.9.5

